tests/parser/cursor-tests.lisp

```diff
@@ -0,0 +1,33 @@
+(fiasco:define-test-package #:coalton-impl/parser/cursor-tests
+  (:use
+   #:cl)
+  (:local-nicknames
+   (#:parser #:coalton-impl/parser)
+   (#:cursor #:coalton-impl/parser/cursor)
+   (#:cst #:concrete-syntax-tree)
+   (#:se #:source-error)))
+
+(in-package #:coalton-impl/parser/cursor-tests)
+
+(defun make-cursor (string)
+  (with-input-from-string (stream string)
+    (parser:with-reader-context stream
+      (cursor:make-cursor (parser:maybe-read-form stream parser::*coalton-eclector-client*)))))
+
+(deftest read-forward ()
+  (let ((c (make-cursor "(1 2 3)")))
+    (is (cst:consp (cursor:cursor-value c)))
+    (is (eq 1 (cursor:next c)))
```

Comment:
eql with numbers (that may also inadvertently not be a number), not eq

Reply:
fixed